TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/utils.py

```diff
@@ -32,3 +36,21 @@ class ExitCode(Enum):
 def generate_log_file_path(topic: str) -> str:
     now = datetime.now().isoformat()
     return f"{now}-{topic}.log"
+
+
+def create_boto3_client(aws_service_name: str, region: Optional[str] = None,
+                        client_options: Optional[ClientOptions] = None):
+    client_config = None
+    if client_options and client_options.user_agent_extra:
+        user_agent_extra_param = {"user_agent_extra": client_options.user_agent_extra}
+        client_config = config.Config(**user_agent_extra_param)
+    return boto3.client(aws_service_name, region_name=region, config=client_config)
+
+
+def append_user_agent_header_for_requests(headers: Optional[dict], user_agent_extra: str):
+    adjusted_headers = dict(headers) if headers else {}
+    if "User-Agent" in adjusted_headers:
+        adjusted_headers["User-Agent"] = f"{adjusted_headers['User-Agent']} {user_agent_extra}"
```

Comment:
The annotation says these aren't covered by a test, but I think this would be a very good condition to have an explicit check on. Can you add a quick one?

Reply:
Added now